tutorials/plugins/gdnative/gdnative-cpp-scons.rst

```diff
@@ -0,0 +1,872 @@
+
+If we are compiling for Windows, then we set our ``env`` to be a new ``Environment`` in which its internal environment is a copy of the operating system's environment variables.
+
+OS-Specific Information
+-----------------------
+
+We'll probably want to give our resulting library a unique name depending on what platform and bit version it's aimed at (32-bit vs. 64-bit). With that in mind, let's also collect the desired bits version.
+
+.. code-block:: python
+
+    platform = ARGUMENTS.get("platform", "windows")
+    bits = ARGUMENTS.get("bits", 64)
+    lib_name = ARGUMENTS.get("name", "libdefault")
+
+    library_file = lib_name + "." + platform + "." + str(bits)
+
+We'll also want to allow the user to decide where to output their various dynamic libraries (one for each platform/bit), so let's also collect the library path.
+
+If we combine the library filename with the platform/bits-specific directory path, then we can determine the full appropriate filename to use for the dynamic library's output location.
+
+.. code-block:: python
+
+    platform = ARGUMENTS.get("platform", "windows")
+    bits = ARGUMENTS.get("bits", 64)
+    lib_name = ARGUMENTS.get("name", "libdefault")
+    lib_path = ARGUMENTS.get("lib", "bin/")
+
+    # Force expected formatting.
+    # Ensures we view Windows-style and filepath-like 
+    # directory paths as typical UNIX directory paths.
+    lib_path = lib_path.split("\\").join("/")
+    if lib_path[-1:] != "/":
+        lib_path += "/"
+
+    platform_dir = ""
+
+    if platform == "osx":
+        platform_dir = "osx" + str(bits) + "/"
+    elif platform == "linux":
+        platform_dir = "x11" + str(bits) + "/"
+    elif platform == "windows":
+        platform_dir = "win" + str(bits) + "/"
+
+    # No need to include the file extension.
+    # scons will determine it automatically based on parameters.
+    library_file = lib_name + "." + platform + "." + str(bits)
+    final_lib_path = lib_path + platform_dir + library_file
+
+Each operating system will also have different flags to pass to the C++ compiler and linker we use.
+
+All of the available options for each platform are really beyond the scope of this tutorial, but you can find more information by examining the compiler options for each individual compiler. In this example, the compiler options specified are related to ``g++`` for OSX/Linux and ``cl`` for Windows.
+
+An overview of the compiler and linker options for ``g++`` can be found here: https://developers.redhat.com/blog/2018/03/21/compiler-and-linker-flags-gcc/
+
+An overview of the compiler options for ``cl`` can be found here: https://msdn.microsoft.com/en-us/library/19z1t1wy.aspx
```

Comment:
While I appreciate that you've easily linked these two resources, I think it's just noise. The reader is here to learn how to build a native library, presumably they know their compiler already.